### PR TITLE
Awicm3 ice thermo cpl fixes

### PR DESCRIPTION
**Pull request by JanStreffing**
This branch is the result of extensive check of the sea ice coupling by Tido, Qiang, Dima Sidorenko, Wolfgang Dorn and others. The changes are confined to routines that are called when FESOM2 is coupled to OpenIFS.

The discussion behind this branch took place in https://github.com/AWI-CM3/project_digital_twin/issues/14

### CODE REVIEW

File: src/ice_thermo_cpl.F90
Comment (koldunovn) [inline src/ice_thermo_cpl.F90:135]: We had before the situation, when people where trying to optimize for `h0`, available in the namelist, but then hardcoded downstream. Will we get into similar trap here? Should we maybe expose those parameters explicitly through `namelist.ice` and comment that they only used in a coupled setting?

**Review by koldunovn — APPROVED** [review]
I am fine with it, have only one comment, that just a suggestion, feel free to consider or ignore it :) I would wait for at least another review before merge.

File: src/ice_thermo_cpl.F90
Comment (dsidoren) [reply src/ice_thermo_cpl.F90:135]: great job, great bug fixes! technically it can be merged as is. @koldunovn shall we introduce namelist.cpl or a section &cpl in one of the namelists for FESOM?

Comment (koldunovn) [reply src/ice_thermo_cpl.F90:135]: Maybe not for this PR, but later I think it's not a bad idea. If you OK with this PR I would merge as is.

Comment (JanStreffing) [reply src/ice_thermo_cpl.F90:135]: I would be ok with that. Any to have the namelist.cpl be a separate development.

**Pull request merged**
